feat: [CDS-72961]: DOC for terraform apply step with encrypt terrafor… (#2301)
harness/developer-hub: docs/continuous-delivery/cd-infrastructure/terraform-infra/run-a-terraform-plan-with-the-terraform-apply-step.md

```
You can use Harness encrypted text for values. See [Add Text Secrets](/docs/platform/Secrets/add-use-text-secrets).

## Encrypt the Terraform Apply JSON outputs

> Currently, this feature is behind the flag, `CDS_ENCRYPT_TERRAFORM_APPLY_JSON_OUTPUT`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.

Reviewer: ```suggestion
Currently, this feature is behind the feature flag, `CDS_ENCRYPT_TERRAFORM_APPLY_JSON_OUTPUT`. Contact [Harness Support](mailto:support@harness.io) to enable this feature.
```

> In Optional Configuration it's required to provide a secret manager for the "Encrypt json output" field.

Reviewer: ```suggestion
In **Optional Configuration** of the Terraform Apply step, you must provide a secret manager for the **Encrypt json output** field.
```

> In **Optional Configuration** of the Terraform Apply step, you must provide a secret manager for the **Encrypt json output** field.

Reviewer: ```suggestion
In the Terraform Apply step's **Optional Configuration**, you must select a secret manager for the **Encrypt json output** field to encrypt and provide the Terraform JSON output as a secret expression.
```

This feature encrypts and provides the Terraform JSON output as a secret expression in the Terraform Apply step. 

Here's an example of encrypted Terraform JSON output as a secret expression: `<+secrets.getValue("encrypted_tf_output_id_1")>`.

```
{
"test-output-name1": {
"sensitive": false,
"type": "string",
"value": "test-output-value1"
}
```

> This secret expression can be used in other steps that can parse, validate,and extract the required output values from the JSON.

Reviewer: ```suggestion
This secret expression can be used in other steps that can parse, validate, and extract the required output values from the JSON.
```
